crates/sui-data-ingestion/src/bin/indexer.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use clap::*;
+use std::path::PathBuf;
+use sui_data_ingestion::{DynamoDBProgressStore, S3Worker};
+use sui_data_ingestion::{IndexerExecutor, WorkerPool};
+use tokio::signal;
+use tokio::sync::oneshot;
+
+#[derive(Parser, Clone, Debug)]
+pub struct IndexerConfig {
+    #[clap(long, global = true)]
+    path: PathBuf,
+    #[clap(long)]
+    aws_access_key: String,
+    #[clap(long)]
+    aws_secret_key: String,
+    #[clap(long)]
+    aws_region: String,
+    #[clap(long)]
+    bucket_name: String,
+    #[clap(long)]
+    table_name: String,
+}
+
+fn setup_env(exit_sender: oneshot::Sender<()>) {
+    let default_hook = std::panic::take_hook();
+
+    std::panic::set_hook(Box::new(move |panic| {
+        default_hook(panic);
+        std::process::exit(12);
+    }));
+
+    tokio::spawn(async {
+        signal::ctrl_c()
+            .await
+            .expect("Failed to install Ctrl+C handler");
+        exit_sender
+            .send(())
+            .expect("Failed to gracefully process shutdown");
+    });
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let (exit_sender, exit_receiver) = oneshot::channel();
+    let config = IndexerConfig::parse();
+    setup_env(exit_sender);
+
+    let progress_store = DynamoDBProgressStore::new(
+        &config.aws_access_key,
+        &config.aws_secret_key,
+        config.aws_region.clone(),
+        config.table_name,
+    )
+    .await;
+    let worker = S3Worker::new(
+        &config.aws_access_key,
+        &config.aws_secret_key,
+        config.aws_region,
+        config.bucket_name,
+    )
+    .await;
+    let mut executor = IndexerExecutor::new(progress_store);
```

Comment:
This is a reference example of the indexer binary, not the actual implementation we're planning to use. Feel free to skip the review of this file.
The main implementation will be YAML-based, enabling the execution of multiple workflows on the same host, along with customization for their concurrency settings.

This is a closer version to what will be used by the community. External developers will need to implement a custom worker (impl. `Worker` trait) or choose from one of the predefined workflows and run the executor